version: "3"

services:
  jupyterlab:
    image: ${IMAGE}
    restart: unless-stopped
    volumes:
      - ./data/work:/home/jovyan/work
      - ./data/.jupyter:/home/jovyan/.jupyter
      - ./data/.ssh:/home/jovyan/.ssh
      - ./jupyter_server_config.json:/home/jovyan/.jupyter/jupyter_server_config.json
    environment:
      - GRANT_SUDO=yes
    security_opt:
      - no-new-privileges:true
    labels:
      my.zone: zone1
      traefik.enable: true
      traefik.http.routers.jupyterlab.rule: Host(`${SUB_DOMAIN}.${SERVER_DOMAIN}`)
      traefik.http.routers.jupyterlab.entrypoints: websecure
      traefik.http.services.jupyterlab.loadbalancer.server.port: ${SERVICE_PORT}
      homepage.group: ${HOMEPAGE_GROUP}
      homepage.name: ${HOMEPAGE_NAME}
      homepage.icon: ${HOMEPAGE_ICON}
      homepage.href: https://${SUB_DOMAIN}.${SERVER_DOMAIN}
      homepage.description: ${HOMEPAGE_DESCRIPTION}

networks:
  default:
    name: zone1
    external: true
